### PR TITLE
Add automatically-generated script to manage user env variables in Windows

### PR DESCRIPTION
**Pull request by traversaro**
These powershell scripts can be used to automatically add or remove the persistent "User"
environmental variables in Windows, automatizing the process that is typically done manually.

Execute `robotology-superbuild/build/install/share/robotology-superbuild/addPathsToUserEnvVariables.ps1` to modify the "User" environmental variables as described in https://github.com/robotology/robotology-superbuild#configure-your-environment-2 , and 
`robotology-superbuild/build/install/share/robotology-superbuild/removePathsFromUserEnvVariables.ps1` to instead cleanup the "User" enviromental variables from any robotology-superbuild path. 

I actually developed this on Linux (powershell on Linux is working quite fine) so I did not checked completely if tricky aspects like paths with spaces, directory separators are properly working, so if anyone can try it would be really helpful. @lrapetti @kouroshD @GiulioRomualdi @Yeshasvitvs 

As in the case of https://github.com/robotology/robotology-superbuild/pull/157 , there is still not documentation because I would like to test the scripts a bit on the real scenario before documenting their use. 

Related to https://github.com/robotology/robotology-superbuild/issues/147 .

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -198,6 +198,11 @@ if(NOT WIN32)
   configure_file_with_cmakeif(${CMAKE_CURRENT_SOURCE_DIR}/cmake/template/setup.sh.in ${CMAKE_BINARY_DIR}/install/share/${PROJECT_NAME}/setup.sh @ONLY)
 else()
   configure_file_with_cmakeif(${CMAKE_CURRENT_SOURCE_DIR}/cmake/template/setup.bat.in ${CMAKE_BINARY_DIR}/install/share/${PROJECT_NAME}/setup.bat @ONLY)
+  configure_file_with_cmakeif(${CMAKE_CURRENT_SOURCE_DIR}/cmake/template/setup.bat.in ${CMAKE_BINARY_DIR}/install/share/${PROJECT_NAME}/setup.bat @ONLY)
```
Comment (diegoferigo) [inline CMakeLists.txt]: Did you duplicate this line by mistake or there is a reason?

Comment (traversaro) [reply CMakeLists.txt]: Copy-paste mistake.

**Comment by lrapetti** [comment]
Some of the paths are created with mixed backslash/fronslash as for example `icub-tests\suits` in the following picture (`YARP_DATA_DIRS_old` are the paths previously setted manually):
path

File: cmake/template/addPathsToUserEnvVariables.ps1.in [hunk 2]
```diff
@@ -0,0 +1,60 @@
+# Automatically generated addPathsToUserEnvVariables.ps1 file for @PROJECT_NAME@
+
+# Set a value to a given "User" enviromental variable
+function Set-ValueToUserEnvVariable ($EnvVariable, $Value, $Verbose=$TRUE) {
+  if ($Verbose) {
+    Write-Host 'Set ' $Value ' to the ' $EnvVariable ' User enviroment variable.'
+  }
+  [System.Environment]::SetEnvironmentVariable($EnvVariable, $newVar, 'User');
```
[the remark this answers is not in the document]
Comment (traversaro) [reply cmake/template/addPathsToUserEnvVariables.ps1.in]: Thanks, addressed in https://github.com/robotology/robotology-superbuild/pull/162/commits/f3745eb8192942888e8dc02eee4c1b8ad50cc537 . In the future feel free to use the "Suggested Changes" feature of GitHub to suggest a change (https://github.blog/changelog/2018-10-16-suggested-changes/) like this, it is quite handy.

**Comment by traversaro** [comment]
> Some of the paths are created with mixed backslash/fronslash as for example `icub-tests\suits` in the following picture

That's exactly what I wanted to check. Apparently CMake generates the `@PROJECT_SOURCE_DIR@` with Unix-style directory separators (`/`). If everything works fine with both `\` or `/`, I think the easiest option is to just generate all the paths with `/`. What do you think @lrapetti ? If you want to do the modifications yourself, I added you as a collaborator on my robotology-superbuild's fork.

**Comment by traversaro** [comment]
> Anyway, to be more consistent I agree on having `/` paths, I can proceed with those changes.

Great, thanks.

**Comment by traversaro** [comment]
If you verified that this is working fine both for adding and removing env variables, I would be happy to merge. We can add documentation in a separate PR.

**Comment by lrapetti** [comment]
> If you verified that this is working fine both for adding and removing env variables, I would be happy to merge. We can add documentation in a separate PR.

I just tested adding and removing and it works. For me you can procede with merging.

In the following pictures you can see the new created environment variables (identified with `new`), and compare them with those manually set:
env_var_1_edit
env_var_2_edit

**Comment by traversaro** [comment]
Thanks a lot @lrapetti .

**Pull request merged**